### PR TITLE
add Cardano support

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: gowrapper/main.go [hunk 1]
```diff
@@ -205,6 +206,10 @@ type jsDevice struct {
 	readChan chan<- []byte
 }
 
+func (device *jsDevice) Version() string {
```
Comment (x1ddos) [inline gowrapper/main.go:209]: this is new... but how is it related to cardano?

File: demo/demo.js [hunk 2]
```diff
@@ -364,3 +368,77 @@ document.getElementById("btcSignMultisig").addEventListener("click", async () =>
...
+                 },
+             },
+        }],
+        fee,
+        ttl,
+        certificates: [
+            { "stakeRegistration": {
+                "keypath": getKeypathFromString("m/1852'/1815'/0'/2/0"),
+            } },
+            { "stakeDelegation": {
```
Comment (x1ddos) [inline demo/demo.js:436]: how does this work? looks like `func (cert *cardanoCertificate) toCertificate()` will ignore delegation if registration is set

File: gowrapper/main.go (hunk 1 above)
Comment (benma) [reply gowrapper/main.go:209]: Ah, it is needed in Adalite. I split it into a new commit.

File: demo/demo.js (hunk 2 above)
Comment (benma) [reply demo/demo.js:436]: This is an list of certificates, each of which can be one of the different certificate types. This tx has two certificates, registration and delegation.

In other words, `toCertificate()` is per certificate, and there can be many.

Comment (x1ddos) [reply demo/demo.js:436]: yeah, i should've commented in the `toCertificate`. what if i do this:

```js
certificates: [
  {
     stakeRegistration: { ... },
     stakeDelegation: { ... },
  }
]
```

- the `toCertificate` will ignore `stakeDelegation` part
- but i will be thinking all is good
- the result is actually not what i'd probably expect

i think `toCertificate` should return an error if more than one field is specified in a single cert object.

or change types to make them explicit `stakeRegistrationCert`, `stakeDelegationCert` and so on. you can't have multiple stakeRegistration certs in a single tx anyway, can you? if so, maybe an object is a better fit for `certificates`?

```js
certificates: {
  registration: {keypath: ...},
  delegation: {keypath: ..., poolKeyhash: ...}
}
```

File: src/bitbox02.js [hunk 3]
```diff
@@ -270,6 +270,12 @@ export class BitBox02API {
         }
     }
 
+    /**
+     * @return the firmware version.
+     */
+    version() {
```
Comment (x1ddos) [inline src/bitbox02.js:276]: add this to the methods.md docs?

**Pull request merged**